src/librustdoc/html/templates/type_layout.html

```diff
@@ -28,13 +28,15 @@ <h2 id="layout" class="section-header"> {# #}
         {% endfor %}
     </ul>
     {% endif %}
-    {# This kind of layout error can occur with valid code, e.g. if you try to
-    get the layout of a generic type such as `Vec<T>`. #}
+        {# This kind of layout error can occur with valid code.
+        It can happen for various reasons, e.g., if you try to get
+        the layout of an array with a length that is not a constant
+        and it has not a generic parameter such as
+        `[(); size_of::<<() as Trait>::Assoc>()]`. #}
```

Comment:
Let's use a simpler example and something that can actually appear in rustdoc:

```suggestion
        {# This kind of layout error can occur with valid code,
        for example if there are trivial bounds:
        `struct Foo(str, str) where str: Sized;`. #}
```